File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        ans=[-1,-1]
        l=0
        r=len(nums)-1
        while l<=r:
            mid=l+(r-l)//2
            if nums[mid]==target:
                ans[0],ans[1]=mid,mid
                r=mid-1
            if nums[mid]<target:
                l=mid+1
            if nums[mid]>target:
                r=mid-1
        if ans[1]>-1:
            while ans[1]<len(nums)-1 and nums[ans[1]+1]==target:
                ans[1]+=1
        return ans

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int []ans={-1,-1};
        int s=0;
        int e=nums.length-1;
        while(s<=e){
            int mid=(s+e)/2;
            if(nums[mid]==target){
                ans[0]=mid;
                ans[1]=mid;
                break;
            }else if(nums[mid]<target){
                s=mid+1;
            }else{
                e=mid-1;
            }
        }
            if(ans[0]!=-1){
                while(ans[0]>0 && nums[ans[0]-1]==nums[ans[0]]){
                    ans[0]-=1;            
                }
            }
        if(ans[1]!=-1){
            while(ans[1]<nums.length-1 && nums[ans[1]]==nums[ans[1]+1]){
                ans[1]+=1;
            }
        }
        return ans;
    }
}